### ISSUE TITLE
XOWA developer backlog: 2018-12

### ISSUE THREAD

**Issue by gnosygnu** [gnosygnu/xowa#310]
I thought it might be helpful to come up with a general backlog issue to get a better idea of timelines.

At current, it stands as the items below. I'll update them as statuses change.

Thanks.

---

* Fixing a long-standing {{#tag}} bug with nested HTML
* Looking at some of the recent nowiki issues 
* Disabling the HZIP from the HTML databases
* Adding some code to <templatestyles> to cache css when the same block is used multiple times on the page (en.w:Henry_David_Thoreau)
* Generating a new English Wikipedia HTML build 
* Generating a new XOWA build

**Comment by desb42** [comment]
With regard to your 4th point (caching css),
I can see the css and it's deduping in the wiki code, but my version of xowa does not even generate css to be deduped.
However, in en.wikiversity I can see the effect clearly - take a look at en.wikiversity.org/wiki/WikiJournal_Preprints/Binary_search_algorithm
In the Citations section are a lot of exposed <style> elements
mediawiki dedups them

I was going to create an issue for this, but you seem to have this covered

**Comment by gnosygnu** [comment]
> In the Citations section are a lot of exposed <style> elements

Yeah, I'm assuming this doesn't affect how the page is rendered? Just a lot of extra HTML in the source view?

Will comment on your other issues later. Thanks!

**Comment by desb42** [comment]
On some further investigations taking
en.wikiversity.org/wiki/WikiJournal_Preprints/Binary_search_algorithm

If I generate with a pass through jtidy I get
![bsa_withtidy](https://user-images.githubusercontent.com/861631/50535485-6dd60380-0b42-11e9-8965-11c2a3d8949e.jpg)
Without
![bsa_notidy](https://user-images.githubusercontent.com/861631/50535491-7c241f80-0b42-11e9-9184-edd2951722ed.jpg)

This is what I mean by exposed \<style> elements

I think that the \<style> elements should not be 'exposed' (then tidy would not eliminate them).
Once that is sorted, then the dedup needs to be done (or at the same time)

The real one looks like
![bsa_wiki](https://user-images.githubusercontent.com/861631/50535644-f1dcbb00-0b43-11e9-9aec-bbfb1145b74f.jpg)

**Comment by desb42** [comment]
you have probable seen this link also https://www.mediawiki.org/wiki/Parsing/Replacing_Tidy

**Comment by gnosygnu** [comment]
Thanks. I'll go thru the link more later, but for now, I'll have to address RemexHtml issues as they come up. I don't know how to handle it on a larger scale as it's PHP and I don't have the time / inclination to port it to Java.

Regarding the en.wikiversity.org images above and the multi-column references: This behavior is actually related to a recent change to the Cite extension. If you're curious, see the following links:
* Wiki: https://www.mediawiki.org/wiki/Contributors/Projects/Columns_for_references
* Code: https://github.com/wikimedia/mediawiki-extensions-Cite/blob/master/includes/Cite.php#L784

I'll have to change XOWA's cite code to handle the new responsive behavior

Finally, I pushed a change for better handling of &lt;templatestyles>. See #314
